### PR TITLE
Add artifact Dependencies inside artifactsFolder to be deployed

### PR DESCRIPTION
**Pull request by louagej**
All dependencies are included in the artifactsFolder during the build task in the CICD
But the deployment only picks the `Apps-*.*.*.*` files from this folder.
I added the `Dependencies-*.*.*.*` app files as well

### CODE REVIEW

**Review by mazhelez — DISMISSED** [review]

File: Actions/Deploy/Deploy.ps1 [hunk 1]
```diff
@@ -48,6 +48,8 @@ if (Test-Path $artifactsFolder -PathType Container) {
         $refname = "$ENV:GITHUB_REF_NAME".Replace('/','_')
         Write-Host "project '$project'"
         $projectApps = @((Get-ChildItem -Path $artifactsFolder -Filter "$project-$refname-$($buildMode)Apps-*.*.*.*") | ForEach-Object { $_.FullName })
+        $projectDependencies = @((Get-ChildItem -Path $artifactsFolder -Filter "$project-$refname-$($buildMode)Dependencies-*.*.*.*") | ForEach-Object { $_.FullName })
+        $projectApps = $projectApps + $projectDependencies
         if (!($projectApps)) {
             if ($project -ne '*') {
                 throw "There are no artifacts present in $artifactsFolder matching $project-$refname-$($buildMode)Apps-<version>."
```
[the remark this answers is not in the document]
Comment (mazhelez) [reply Actions/Deploy/Deploy.ps1]: Actually, could there be dependencies, but no apps? 

It's better to keep the original check _before_ adding the dependencies to the list of apps to deploy.

[the remark this answers is not in the document]
Comment (louagej) [reply Actions/Deploy/Deploy.ps1]: Restored the original check before adding the dependencies to the list of apps to deploy

**Comment by freddydk** [comment]
Before approving this, we need to investigate and answer a few things:
- Do we want to same behavior on production environments and sandbox environments?
- What if the app already is installed in an earlier version - should we upgrade?
- What if the app already is installed in a newer version - should we fail?
- What if the dependencies are runtime packages (or symbols) of appsource apps - should we then try to install the appsource app from AppSource?
- What if the dependency is an appsource app and we are using Per Tenant Extension installation (not Dev scope) - should we then install the app from appsource?

**Comment by freddydk** [comment]
> I added some points with my opinion. So don't take these as hard rules, but rather as a train of thought 🚂💭
> 
> * Do we want the same behavior in production environments and sandbox environments?
>   > * I presume yes, it should have the same behavior.
> * What if the app is already installed in an earlier version - should we upgrade?
>   > * Skip the install of this dependent app. If the administrator of this environment wants the dependency to update, that's their role. If the administrator doesn't want to update the dependent app, and the deployment fails because of this, it should be a discussion between the partner providing the app and the customer on how to deal with outdated apps.
> * What if the app is already installed in a newer version - should we fail?
>   > * Skip the install of this dependent app. Same as above with the earlier version. It should be a discussion between the partner providing the app and the customer on how to deal with outdated/newer apps.
> * What if the dependencies are runtime packages (or symbols) of AppSource apps - should we then try to install the AppSource app from AppSource?
>   > * In my opinion, an AppSource app should take precedence over the provided app in the artifact.
> * What if the dependency is an AppSource app and we are using Per Tenant Extension installation (not Dev scope) - should we then install the app from AppSource?
>   > * In my opinion, install the app from AppSource.

So, basically:
- If the App (from dependencies) is a PTE:
   - If it is installed in a newer version - leave it and see if it works - agree
   - If it is installed in an older version - I think we should install the newer version (kind of like if it isn't installed). If not, then eventually this will fail and cause people to need to find this dependency and install manually.

- If the App (from dependencies) is an AppSource App:
   - If it isn't already installed - install it from AppSource - never push to dev scope - I agree on this
   - If is is installed in an older version (than specifies in dependencies) - shouldn't we update the AppSource app?

Should we have a setting under DeployTo<environment> which dictates this behavior?`

All in all - it isn't done by treating dependencies as they were the same as apps - apps might be installed in Dev Scope - dependencies shouldn't. Apps will always upgrade etc etc

**Comment by louagej** [comment]
I agree on your last comments but guess then just adding these dependency apps to this array of apps as provided in the pull request isn't enough.

**Comment by freddydk** [comment]
> I agree on your last comments but guess then just adding these dependency apps to this array of apps as provided in the pull request isn't enough.

Exactly - I think it requires a bit more

**Comment by freddydk** [comment]
I believe this is what would be needed: https://github.com/microsoft/AL-Go/pull/1225
Haven't tested it yet - but I do think this should do what's needed.
@louagej - please have a look and let me know if you see something wrong.

**Comment by louagej** [comment]
@freddydk,
This looks perfect to me!
I started yesterday by trying to modify the PR according to your comments, but I couldn’t have come up with something this good… #1225
🚀

**Comment by freddydk** [comment]
> @freddydk, This looks perfect to me! I started yesterday by trying to modify the PR according to your comments, but I couldn’t have come up with something this good… #1225 🚀

Just tested my PR by using Update AL-Go System Files with freddydk/AL-Go@issue1215 - seems to work just fine. Feel free to test it yourself.

Will close this PR and change the other PR from draft to review ready later.

**Pull request closed without merging**